build-tools/tasks/test-utils.js

```diff
@@ -57,12 +105,22 @@ function generateIndexFileContent(testUtilType, testUtilMetaData) {
     ...testUtilMetaData.map(({ componentName }) => {
       const wrapperName = toWrapper(componentName);
       // language=TypeScript
-      return `ElementWrapper.prototype.find${componentName} = function(selector) {
-          const rootSelector = \`.$\{${wrapperName}.rootSelector}\`;
-          // casting to 'any' is needed to avoid this issue with generics
-          // https://github.com/microsoft/TypeScript/issues/29132
-          return (this as any).findComponent(selector ? appendSelector(selector, rootSelector) : rootSelector, ${wrapperName});
-      };`;
+      return `
+          ElementWrapper.prototype.find${componentName} = function(selector) {
+            const rootSelector = \`.$\{${wrapperName}.rootSelector}\`;
+            return this.findComponent(selector ? appendSelector(selector, rootSelector) : rootSelector, ${wrapperName});
+          };
+
+          ElementWrapper.prototype.findAll${pluralize(componentName)} = function(selector) {
+            const rootSelector = \`.$\{${wrapperName}.rootSelector}\`;
+            return this.findAllComponents(selector ? appendSelector(selector, rootSelector) : rootSelector, ${wrapperName});
+          };
+
+          ElementWrapper.prototype.find${componentName}ByTestId = function(testId) {
+            const selector = \`.\${${wrapperName}.rootSelector}[data-testid="\${testId}"]\`;
```

Comment:
I think we need to escape the `testId` value before inserting it into the string. Otherwise, a testId that contains quotes (e.g. `ab"c"`) or other special characters will break this selector. The `CSS.escape` function seems like the right one to use: https://developer.mozilla.org/en-US/docs/Web/API/CSS/escape_static

Reply:
Great catch 💪 Fixed and added a test case for it.